### ISSUE TITLE
[Content Types] : Select field must not allow incorrect values

### ISSUE THREAD

**Issue by jcastro-dotcms** [dotCMS/core#21210]
Reported via Support: [105576](https://dotcms.zendesk.com/agent/tickets/105576)
Reproducible as of dotCMS 21.06.3, up to latest master (21.10)

The `Help` -- question mark -- icon in any **Select** field edit modal shows the following information:
```
Allows a user to select only one option from a list of options.
 
Set options to display like this:
Text|Value
Canada|CA
Mexico|MX
USA|US
 
Note: If you want to use True/False as your data type, you must use your native db representation for true and false, e.g.
True|1
False|0.
```
However, the system is **NOT** making sure that the user follows such rules. This is causing problems when rendering, and filling the log with large stack traces whose error message is not easy to interpret. For example:

**To Reproduce**
1. Create a test Content Type `My CT` with two fields:
1.1 Text field named `Title`.
1.2 Selector field named `My Selector` with these values -- notice how the input has no pipes to differentiate the display value and real value, and the presence of double quotes:
```
Default
Additional "Get Started" Button
Referrals
```
2. Create/update a Template and a Container to print out the title of `My CT` contents.
3. Create a new contentlet of type `My CT`, and create a test HTML Page so that you can add it.
4. The page renders. However, the following error is being logged:
```
09:33:54.321  WARN  resource.ResourceManagerImpl - ResourceManager.getResource() parse exception @ url:GET//{SITE-NAME}/api/v1/page/render/{PAGE-URL} | lang:1 | ip:127.0.0.1 | Admin:true | start:10-28-2021 08:15:28 CST  ref:http://localhost:8080/dotAdmin/  ?language_id=1&host_id=d6447a46-2663-4661-9dba-1c1660b8c5e5
com.dotcms.rendering.velocity.events.PreviewEditParseErrorException: Encountered "\"Get Started\"" at /PREVIEW_MODE/f52c8839-f44c-4db9-8137-5713a06fff4f_1.contentlet[line 1, column 1339]
Was expecting one of:
    <RPAREN> ...
    <WHITESPACE> ...
    "-" ...
    "+" ...
    "*" ...
    "/" ...
    "%" ...
    <LOGICAL_AND> ...
    <LOGICAL_OR> ...
    <LOGICAL_LT> ...
    <LOGICAL_LE> ...
    <LOGICAL_GT> ...
    <LOGICAL_GE> ...
    <LOGICAL_EQUALS> ...
    <LOGICAL_NOT_EQUALS> ...
    
	at com.dotcms.rendering.velocity.events.ExceptionHandlerParseErrorException.handle(ExceptionHandlerParseErrorException.java:37) ~[dotcms_21.06.3_e0afd30.jar:?]
	at com.dotcms.rendering.velocity.events.ExceptionHandlerParseErrorException.handle(ExceptionHandlerParseErrorException.java:1) ~[dotcms_21.06.3_e0afd30.jar:?]
	at org.apache.velocity.Template.process(Template.java:163) ~[dotcms_21.06.3_e0afd30.jar:?]
...
...
```
And a notification in the UI as well:
<img width="300" alt="Screen Shot 2021-10-25 at 11 00 46 PM" src="https://user-images.githubusercontent.com/8779585/139303186-5782782b-4a51-4c0e-b2fa-e0d42d7dc781.png">

**Expected behavior**
The application must let the user know that the inout values are not correct.

**Additional context**
Users can escape the double quotes by using the `$esc.` and this would get rid of the errors in the log. However, in the UI, the selector is rendered by JavaScript, so the display value is rendered like this: `New $esc.qGet Started$esc.q Button` which is not easy to read for most content authors.

We might need to validate this in the [ContentletLoader](https://github.com/dotCMS/core/blob/master/dotCMS/src/main/java/com/dotcms/rendering/velocity/services/ContentletLoader.java) class as well.

**Comment by jcastro-dotcms** [comment]
@dsilvam @freddyucv is the pipe character required when adding values to a **Select** field? If it is, we need to enforce an additional validation for it

**Comment by bryanboza** [comment]
Fixed, tested on release-21.12 and now we are not getting the error mentioned above, We will move the card to next release in order to finish the validation requested by @jcastro-dotcms in the previous comment

**Comment by wezell** [comment]
This is working as expected.  If there is no pipe, we assume the label and the value are the same.
